extensions/sql-database-projects/src/tools/buildHelper.ts

```diff
@@ -51,29 +59,30 @@ export class BuildHelper {
 			await fs.mkdir(this.extensionBuildDir);
 		}
 
-		const buildfilesPath = await this.getBuildDirPathFromMssqlTools();
-		buildFiles.forEach(async (fileName) => {
+		// download the Microsoft.Build.Sql sdk nuget
+		const httpClient = new HttpClient();
+		const nugetPath = path.join(this.extensionBuildDir, `${fullSdkName}.nupkg`);
+		await httpClient.download(microsoftBuildSqlUrl, nugetPath, outputChannel);
+
+		// extract the files from the nuget
+		await extractZip(nugetPath, { dir: path.join(this.extensionDir, buildDirectory, sdkName) });
+
+		outputChannel?.appendLine(constants.extractingDacFxDlls);
+
+		// copy the dlls and targets file to the BuildDirectory folder
+		const buildfilesPath = path.join(this.extensionDir, buildDirectory, sdkName, 'tools', 'netstandard2.1');
+
+		for (const fileName of buildFiles) {
 			if (await (utils.exists(path.join(buildfilesPath, fileName)))) {
 				await fs.copyFile(path.join(buildfilesPath, fileName), path.join(this.extensionBuildDir, fileName));
 			}
-		});
-		this.initialized = true;
-	}
-
-	/**
-	 * Gets the path to the SQL Tools Service installation
-	 * @returns
-	 */
-	private async getBuildDirPathFromMssqlTools(): Promise<string> {
-		try {
-			if (utils.getAzdataApi()) {
-				return (vscode.extensions.getExtension(mssql.extension.name)?.exports as mssql.IExtension).sqlToolsServicePath;
-			} else {
-				return (vscode.extensions.getExtension(vscodeMssql.extension.name)?.exports as vscodeMssql.IExtension).sqlToolsServicePath;
-			}
-		} catch (err) {
-			throw new Error(errorFindingBuildFilesLocation(err));
 		}
+
+		// cleanup extracted folder and nuget
```

Comment:
May be worth trying to look in the nuget cache locations first to avoid having to download if those exist. 

Or at least I'm not sure we want to clean this up here so we don't have to redownload and re-extract the DLLs each time.

Reply:
added a TODO to look into checking nuget caches in a separate PR. I updated this to no longer redownload on each ADS restart, only if the nuget version does not match the expected one.